### PR TITLE
Add introduction  of  dm relay  log

### PR DESCRIPTION
**Pull request by sunzhaoyang**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [x] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dm/dm-relay-log.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
+---
+title: DM relay log 
+summary: 介绍 DM relay log 的作用及原理
+---
+
+
```
Comment (sunzhaoyang) [inline dm/dm-relay-log.md:6]:
```suggestion
```

File: TOC.md [hunk 2]
```diff
@@ -348,6 +348,7 @@
         - 架构
           - [DM 架构简介](/dm/dm-arch.md)
           - [DM-worker 说明](/dm/dm-worker-intro.md)
+          - [relay log](/dm/dm-relay-log.md)
```
Comment (ran-huang) [inline TOC.md:351]: 这个文档和 https://docs.pingcap.com/zh/tidb/dev/relay-log 是什么关系，看起来重复了？

Comment (sunzhaoyang) [reply TOC.md:351]: 更加细致的解释。我合并成一篇吧

File: dm/relay-log.md [hunk 3]
```diff
@@ -10,91 +10,19 @@ DM (Data Migration) 工具的 relay log 由若干组有编号的文件和一个
 
 在启用 relay log 功能后，DM-worker 会自动将上游 binlog 迁移到本地配置目录（若使用 TiUP 部署 DM，则迁移目录默认为 `<deploy_dir> / <relay_log>`）。本地配置目录 `<relay_log>` 的默认值是 `relay-dir`，可在[上游数据库配置文件](/dm/dm-source-configuration-file.md)中进行修改。自 v5.4.0 版本起，你可以在 [DM-worker 配置文件](/dm/dm-worker-configuration-file.md)中通过 `relay-dir` 配置本地配置目录，其优先级高于上游数据库的配置文件。
 
-DM-worker 在运行过程中，会将上游 binlog 实时迁移到本地文件。DM-worker 的 sync 处理单元会实时读取本地 relay log 的 binlog 事件，将这些事件转换为 SQL 语句，再将 SQL 语句迁移到下游数据库。
+![reley](/media/dm/dm-relay-log.png)
```
Comment (ran-huang) [inline dm/relay-log.md:13]:
```suggestion
![relay log](/media/dm/dm-relay-log.png)
```

File: dm/relay-log.md [hunk 4]
```diff
@@ -10,91 +10,19 @@ DM (Data Migration) 工具的 relay log 由若干组有编号的文件和一个
...
-|   `-- relay.meta
-`-- server-uuid.index
-```
-
-- `subdir`：
-
-    - DM-worker 把从上游数据库迁移到的 binlog 存储在同一目录下，每个目录都为一个 `subdir`。
-    - `subdir` 的命名格式为 `<上游数据库 UUID>.<本地 subdir 序列号>`。
-    - 在上游进行 master 和 slave 实例切换后，DM-worker 会生成一个序号递增的新 `subdir` 目录。
+Relay log 的使用场景:
```
Comment (ran-huang) [inline dm/relay-log.md:15]: 文档内的标题结构能否调整成这样：

```
.
└── Relay Log
    ├── Relay log 原理
    │   ├── 适用场景
    │   ├── 限制
    │   └── 目录结构
    ├── 使用 Relay log
    │   ├── 开启/关闭
    │   ├── 查询
    │   ├── 暂停/恢复
    │   └── 清理
    │       ├── 自动数据清理
    │       └── 手动数据清理
    ├── 常见问题
    └── 探索更多
```

File: dm/relay-log.md [hunk 5]
```diff
@@ -10,91 +10,19 @@ DM (Data Migration) 工具的 relay log 由若干组有编号的文件和一个
...
 
-        - 在以上示例中，对于 `7e427cc0-091c-11e9-9e45-72b7c59d52d7.000001` 这一目录，`7e427cc0-091c-11e9-9e45-72b7c59d52d7` 是上游数据库的 UUID，`000001` 是本地 `subdir` 的序列号。
+- Mysql 的存储空间是有限制的，一般都会设置 binlog 的最长保存时间，当上游把 binlog 清除掉之后，如果 DM 还需要对应位置的 binlog 就会拉取失败，导致同步任务出错；
```
Comment (shichun-0415) [inline dm/relay-log.md:17]:
```suggestion
- MySQL 的存储空间是有限制的，一般都会设置 binlog 的最长保存时间，当上游把 binlog 清除掉之后，如果 DM 还需要对应位置的 binlog 就会拉取失败，导致同步任务出错。
```

File: dm/relay-log.md [hunk 6]
```diff
@@ -10,91 +10,19 @@ DM (Data Migration) 工具的 relay log 由若干组有编号的文件和一个
...
+- 若未开启 relay log ，DM 每增加一个同步任务都会在上游建立一条链接用于拉取 binlog，这样会对上游造成比较大的负载，开启 relay log 后同一个上游的多个同步任务可以复用已经拉倒本地的 relay log，这样就减少了对上游的压力；
```
Comment (shichun-0415) [inline dm/relay-log.md:18]:
```suggestion
- 若未开启 relay log，DM 每增加一个同步任务都会在上游建立一条链接用于拉取 binlog，这样会对上游造成比较大的负载。开启 relay log 后，同一个上游的多个同步任务可以复用已经拉到本地的 relay log，这样就减少了对上游的压力。
```

File: dm/relay-log.md [hunk 7]
```diff
@@ -287,9 +215,15 @@ Relay log 迁移的起始位置由如下规则决定：
 
 ## 清理 relay log
 
-因为存在文件读写的检测机制，所以 DM-worker 不会清理正在使用的 relay log，也不会清理当前已有数据迁移任务之后会使用到的 relay log。
+relay log  的清理 DM 提供了两种方式，手动清理和自动清理，需要注意的是这两种清理方法都不会清理活跃的 relay log 。
```
Comment (shichun-0415) [inline dm/relay-log.md:218]:
```suggestion
DM 提供两种清理 relay log 的方式，手动清理和自动清理。需要注意，这两种清理方法都不会清理活跃的 relay log 。
```

File: dm/relay-log.md [hunk 8]
```diff
@@ -287,9 +215,15 @@ Relay log 迁移的起始位置由如下规则决定：
...
 
-Relay log 的数据清理包括自动清理和手动清理这两种方法。
+> **注意：**
+> 
+> 活跃的 relay log  定义：该 relay log  正在被同步任务使用。
+> 
+> 过期的 relay log  定义：该 relay log  文件最后被改动的时间与当前时间差值大于配置文件中的 `expires` 字段。
+> 
+> 活跃的 relay log 当前只在 Syncer Unit 被更新和写入，假设一个为 All 模式的同步任务在全量导出/导入阶段花费了超过数据源 purge 里配置的过期时间，该 relay log  依旧会被清除。
```
Comment (shichun-0415) [inline dm/relay-log.md:226]:
```suggestion
> - 活跃的 relay log：该 relay log  正在被同步任务使用。活跃的 relay log 当前只在 Syncer Unit 被更新和写入，假设一个为 All 模式的同步任务在全量导出/导入阶段花费了超过数据源 purge 里配置的过期时间，该 relay log  依旧会被清除。
> 
> - 过期的 relay log：该 relay log  文件最后被改动的时间与当前时间差值大于配置文件中的 `expires` 字段。
```

File: dm/relay-log.md [hunk 9]
```diff
@@ -170,16 +104,19 @@ Relay log 迁移的起始位置由如下规则决定：
...
-## 查询 relay log
+### 查询 relay log
 
-`query-status -s` 命令可以查询 relay log 的状态。
+使用 `query-status -s` 命令，可以查询 relay log 的状态：
 
 {{< copyable "" >}}
 
 ```bash
 » query-status -s mysql-replica-01
```
Comment (ran-huang) [inline dm/relay-log.md:114]: `»` 是什么意思

File: dm/relay-log.md [hunk 10]
```diff
@@ -370,13 +317,93 @@ deb76a2b-09cc-11e9-9129-5242cf3bb246.000003
...
+|   |-- mysql-bin.000001
+|   `-- relay.meta
+`-- server-uuid.index
+```
+
+- `subdir`：
+
+    - DM-worker 把从上游数据库迁移到的 binlog 存储在同一目录下，每个目录都为一个 `subdir`。
+    - `subdir` 的命名格式为 `<上游数据库 UUID>.<本地 subdir 序列号>`。
+    - 在上游进行 master 和 slave 实例切换后，DM-worker 会生成一个序号递增的新 `subdir` 目录。
```
Comment (ran-huang) [inline dm/relay-log.md:355]: master / slave 是指什么呢？
如果没有特殊的情况，现在一般要求用 source/replica（与 mysql 一致）或者 primary/secondary。

**Comment by ran-huang** [comment]
@sunzhaoyang please confirm the changes in [113d2dd](https://github.com/pingcap/docs-cn/pull/8780/commits/113d2dddd8ad02eba966f092363b55a32601b84e). Rest LGTM

File: dm/relay-log.md [hunk 11]
```diff
@@ -370,13 +317,92 @@ deb76a2b-09cc-11e9-9129-5242cf3bb246.000003
...
```
Comment (sunzhaoyang) [inline dm/relay-log.md:355]:
```suggestion
    - 在上游进行 primary 和 secondary 实例切换后，DM-worker 会生成一个序号递增的新 `subdir` 目录。
```

**Review by ran-huang — APPROVED** [review]

**Comment by ran-huang** [comment]
@Ehco1996 PTAL, thanks. Please approve this PR if you think it's technically correct.

**Review by Ehco1996 — APPROVED** [review]
LGTM

**Pull request merged**